cmds/webboot/webboot.go

```diff
@@ -176,6 +177,36 @@ func (d *DownloadOption) exec(uiEvents <-chan ui.Event, network bool, cacheDir s
 
 	}
 
+	// verify checksum if the distro chosen is supported
+	if _, ok := supportedDistros[entry.Label()]; ok {
+		// check that checksum is available
+		if distro.checksum == "" {
+			accept, err := menu.PromptConfirmation("This distro does not have a checksum. Proceed anyway?", uiEvents)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to prompt confirmation: %s", err)
+			}
+			if !accept {
+				// go back to download menu
+				return &DownloadOption{}, nil
+			}
+		} else {
+			if valid, err := bootiso.VerifyChecksum(fpath, distro.checksum, distro.checksumType); err != nil {
+				return nil, fmt.Errorf("Failed to verify checksum of: %s", err)
+			} else {
+				if !valid {
+					accept, err := menu.PromptConfirmation("Checksum was not correct. Proceed anyway?", uiEvents)
```

Comment:
It might be useful to the user to print the checksum you got the checksum you expected.